onadata/apps/api/management/commands/delete_users.py

```diff
@@ -0,0 +1,126 @@
+        "deleting this account?".format(
+            username,
+            result['projects'],
+            result['forms'],
+            result['submissions']
+            ))
+
+    return user_response
+
+
+def inactivate_user(username, email):
+    """
+    Soft deletes the user termporarily.
+    """
+    try:
+        user = User.objects.get(username=username, email=email)
+        # set inactive status on user account
+        user.is_active = False
+        # append a timestamped suffix to the username
+        # to make the initial username available
+        soft_deletion_time = timezone.now()
+        deletion_suffix = soft_deletion_time.strftime('-deleted-at-%s')
+        user.username += deletion_suffix
+
+        user.save()
+        sys.stdout.write(
+            'User {} deleted successfully.'.format(username))
+        # confirm too that no user exists with provided email
+        if len(User.objects.filter(email=email, is_active=True)) > 1:
+            other_accounts = [
+                user.username for user in User.objects.filter(
+                    email=email)]
+            sys.stdout.write(
+                'User accounts {} have the same '
+                'email address with this User'.format(
+                    other_accounts))
+
+    except User.DoesNotExist:
+        raise CommandError('User {} does not exist.'.format(username))
+
+
+class Command(BaseCommand):
+    """
+    Delete users management command.
+
+    :param user_details:
+    :param user_input:
+
+    Usage:
+    The mandatory arguments are --user_details
+    --user_details username1:email username2:email
+
+    The command defaults the values for the --user_input attribute to False
+    To change this, pass this in with the value True i.e
+    --user_input True
+    """
+    help = 'Delete users'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--user_details', nargs='*')
+
+        parser.add_argument(
+            '--user_input',
+            help='Confirm deletion of user account',
+            default='False'
+        )
+
+    def handle(self, *args, **kwargs):
+        users = kwargs.get('user_details')
+        user_input = kwargs.get('user_input')
+
+        if users:
+            for user in users:
+                username, email = user.split(':')
+                user_response = 'True'
+                if user_input == 'False':
+                    # If the --user_input flag is not provided.
+                    # Get acknowledgement from the user on this
+                    user_response = get_user_object_stats(
+                        username,
+                        email,
+                        user_input
+                    )
+                if user_response == 'True':
+                    inactivate_user(username, email)
+                else:
+                    sys.stdout.write(
+                        'No actions taken')
+        else:
+            raise CommandError('No User Account provided!')
```

Comment:
Since we've chosen to move most of the functionality into separate functions can we move this too ? Possibly changing the `inactivate_user` function to `inactivate_users`; A function that takes a list of `user:email` and deletes the user ?

Reply:
Moved this